src/Task.ts

```diff
@@ -310,22 +320,51 @@ export class Task {
 
             const recurrenceMatch = description.match(Task.recurrenceRegex);
             if (recurrenceMatch !== null) {
-                recurrence = Recurrence.fromText({
-                    recurrenceRuleText: recurrenceMatch[1].trim(),
-                    startDate,
-                    scheduledDate,
-                    dueDate,
-                });
-
+                // Save the recurrence rule, but *do not parse it yet*.
+                // Creating the Recurrence object requires a reference date (e.g. a due date),
+                // and it might appear in the next (earlier in the line) tokens to parse
+                recurrenceRule = recurrenceMatch[1].trim();
                 description = description
                     .replace(Task.recurrenceRegex, '')
                     .trim();
                 matched = true;
             }
 
+            // Match tags from the end of the task description, not to be actually used,
```

Comment:
What does this comment mean by 'not to be actually used'? It's important that all the tags on the line get added to `Task.tags`... (They do, I checked, with tests - but I just wanted to make sure what the comment means...)

Reply:
Rewrote to make it clearer.